app/src/main/java/com/keylesspalace/tusky/db/AccountManager.kt

```diff
@@ -0,0 +1,190 @@
+/* Copyright 2018 Conny Duck
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.db
+
+import android.util.Log
+import com.keylesspalace.tusky.TuskyApplication
+import com.keylesspalace.tusky.entity.Account
+
+/**
+ * This class caches the account database and handles all account related operations
+ * @author ConnyDuck
+ */
+
+class AccountManager {
+
+    var activeAccount: AccountEntity? = null
```

Comment:
I am worried about multi-threading here and in the `accounts` field. For collections we could wrap them in thread-safe decorators, for field you could put it in the `AtomicReference` and delegate to it or we could at least make it `volatile` so every thread will see changes.

Reply:
I encountered a problem with concurrency, but since `getAllAccountsOrderedByActive` returns an immutable list it was fine. But I will make `activeAccount` `volatile` just to be sure.

Reply:
1. It is immutable in Kotlin, but mutable in Java
2. You can mutate underlying collection (not one returned by `getAllAccountsOrderedByActive`) from multiple threads